core/slice.go

```diff
@@ -694,7 +694,46 @@ func (sl *Slice) ConstructLocalBlock(header *types.Header) (*types.Block, error)
 		// in the subordinate manifest.
 		return types.NewBlockWithHeader(header), nil
 	}
-	pendingBlockBody := sl.PendingBlockBody(header.Root())
+	pendingBlockBody := rawdb.ReadCandidateBody(sl.sliceDb, header.Hash())
+	if pendingBlockBody == nil {
+		return nil, ErrBodyNotFound
+	}
+	// Load uncles because they are not included in the block response.
+	txs := make([]*types.Transaction, len(pendingBlockBody.Transactions))
+	for i, tx := range pendingBlockBody.Transactions {
+		txs[i] = tx
+	}
+	uncles := make([]*types.Header, len(pendingBlockBody.Uncles))
+	for i, uncle := range pendingBlockBody.Uncles {
+		uncles[i] = uncle
+		log.Debug("Pending Block uncle", "hash: ", uncle.Hash())
+	}
+	etxs := make([]*types.Transaction, len(pendingBlockBody.ExtTransactions))
+	for i, etx := range pendingBlockBody.ExtTransactions {
+		etxs[i] = etx
+	}
+	subManifest := make(types.BlockManifest, len(pendingBlockBody.SubManifest))
+	for i, blockHash := range pendingBlockBody.SubManifest {
+		subManifest[i] = blockHash
+	}
+	block := types.NewBlockWithHeader(header).WithBody(txs, uncles, etxs, subManifest)
+	if err := sl.validator.ValidateBody(block); err != nil {
+		return block, err
+	} else {
+		return block, nil
+	}
+}
+
+// constructLocalMinedBlock takes a header and construct the Block locally
```

Comment:
Would be nice if the comments explained the difference between the two `Construct...` methods

Reply:
okay, will add

Reply:
added comments on both the functions